pkg/catalog/catalog.go

```diff
@@ -1946,3 +1946,26 @@ func newDifferenceFromEntryDiff(v *EntryDiff) (Difference, error) {
 	diff.Type, err = catalogDiffType(v.Type)
 	return diff, err
 }
+
+func NewUncommittedWriter(writer io.Writer) UncommittedWriter {
```

Comment:
I would return a pointer to an implementation in case we like assign the return value to an interface.